experiment/measurer.py

```diff
@@ -67,13 +68,158 @@ def exists_in_experiment_filestore(path: pathlib.Path) -> bool:
                               must_exist=False).retcode == 0
 
 
+def get_summary_file_path(fuzzer: str, benchmark: str, trial_num: int):
+    """Return summary file path."""
+    benchmark_fuzzer_trial_dir = experiment_utils.get_trial_dir(
+        fuzzer, benchmark, trial_num)
+    work_dir = experiment_utils.get_work_dir()
+    measurement_dir = os.path.join(work_dir, 'measurement-folders',
+                                   benchmark_fuzzer_trial_dir)
+    summary_file_path = os.path.join(measurement_dir, 'reports',
+                                     'cov_summary.json')
+    return summary_file_path
+
+
+def get_fuzzer_benchmark_key(fuzzer: str, benchmark: str):
+    """Return the key in coverage dict for a pair of fuzzer-benchmark."""
+    return fuzzer + ' ' + benchmark
+
+
+def get_all_fuzzer_names(experiment: str):
+    """Return all fuzzer names for the experiment."""
+    fuzzers = [
+        fuzzer_tuple[0] for fuzzer_tuple in db_utils.query(models.Trial.fuzzer).
+        distinct().filter(models.Trial.experiment == experiment)
+    ]
+    return fuzzers
+
+
+def get_all_benchmark_names(experiment: str):
+    """Return all benchmark names for the experiment."""
+    benchmarks = [
+        benchmark_tuple[0]
+        for benchmark_tuple in db_utils.query(models.Trial.benchmark).distinct(
+        ).filter(models.Trial.experiment == experiment)
+    ]
+    return benchmarks
+
+
+def get_trial_ids(experiment: str, fuzzer: str, benchmark: str):
+    """Get ids of all finished trials for a pair of fuzzer and benchmark."""
+    trial_ids = [
+        trial_id_tuple[0]
+        for trial_id_tuple in db_utils.query(models.Trial.id).distinct().filter(
+            models.Trial.experiment == experiment, models.Trial.fuzzer ==
+            fuzzer, models.Trial.benchmark == benchmark,
+            ~models.Trial.preempted)
+    ]
+    return trial_ids
+
+
+def eliminate_set(dictionary):
+    """Transform the set structure to list so we can use json.dumps."""
+    for key in dictionary:
+        dictionary[key] = list(dictionary[key])
+
+
+def store_cov_data(experiment: str):
+    """Generate the specific coverage data and store in cloud bucket."""
+    logger.info('Start storing coverage data')
+    with multiprocessing.Pool() as pool, multiprocessing.Manager() as manager:
+        q = manager.Queue()  # pytype: disable=attribute-error
+        try:
+            covered_regions = get_all_covered_region(experiment, pool, q)
+            json_src_dir = get_experiment_folders_dir()
+            json_src = os.path.join(json_src_dir, 'covered_regions.json')
+            with open(json_src, 'w') as src_file:
+                json.dump(covered_regions, src_file)
+            json_dst = exp_path.filestore(json_src)
+            filestore_utils.cp(json_src, json_dst, expect_zero=False)
+        except Exception:  # pylint: disable=broad-except
```

Comment:
Why do need broad-except ? Do we even need it at all ? covered_regions.json should be alwasys valid ?